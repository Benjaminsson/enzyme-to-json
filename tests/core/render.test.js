/* eslint-env jest */

import React from 'react';
import Enzyme, {render} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import renderToJson from '../../src/render';
import {BasicPure, BasicWithAList} from './fixtures/pure-function';
import {BasicClass, ClassWithNull} from './fixtures/class';

Enzyme.configure({adapter: new Adapter()});

it('converts basic pure render', () => {
  const rendered = render(
    <BasicPure className="pure">
      <strong>Hello!</strong>
    </BasicPure>,
  );

  expect(renderToJson(rendered)).toMatchSnapshot();
});

it('converts basic class render', () => {
  const rendered = render(
    <BasicClass className="class">
      <strong>Hello!</strong>
    </BasicClass>,
  );

  expect(renderToJson(rendered)).toMatchSnapshot();
});

it('renders the whole list', () => {
  const wrapper = render(<BasicWithAList />);
  expect(renderToJson(wrapper.find('ul'))).toMatchSnapshot();
});

it('handles a component which returns null', () => {
  const rendered = render(<ClassWithNull />);

  expect(renderToJson(rendered)).toMatchSnapshot();
});

it('renders multiple elements as a result of find', () => {
  const rendered = render(<BasicWithAList />);
  expect(renderToJson(rendered.find('li'))).toMatchSnapshot();
});
