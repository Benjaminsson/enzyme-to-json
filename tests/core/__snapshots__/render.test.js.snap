// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts basic class render 1`] = `
<div
  class="basic-class class"
>
  <div
    class="group"
    id="group-id"
  >
    <span>
      <strong>
        Hello!
      </strong>
    </span>
    <span
      class="empty"
    />
  </div>
</div>
`;

exports[`converts basic pure render 1`] = `
<div
  class="basic-pure pure"
>
  <div
    class="group"
    id="group-id"
  >
    <span>
      <strong>
        Hello!
      </strong>
    </span>
    <span
      class="empty"
    />
  </div>
</div>
`;

exports[`converts class components with render returning top level arrays 1`] = `
Array [
  <div
    class="test"
  >
    Test
  </div>,
  <div
    class="test2"
  >
    Test 2
  </div>,
  <div
    class="child"
  >
    <strong>
      Hello!
    </strong>
  </div>,
]
`;

exports[`converts function components with render returning top level arrays 1`] = `
Array [
  <div
    class="test"
  >
    Test
  </div>,
  <div
    class="test2"
  >
    Test 2
  </div>,
  <div
    class="child"
  >
    <strong>
      Hello!
    </strong>
  </div>,
]
`;

exports[`handles a component which returns null 1`] = `null`;

exports[`renders multiple elements as a result of find 1`] = `
Array [
  <li>
    0
  </li>,
  <li>
    1
  </li>,
]
`;

exports[`renders the whole list 1`] = `
<ul>
  <li>
    0
  </li>
  <li>
    1
  </li>
</ul>
`;
